File: day01/part2.swift
var list1: [Int] = []
var list2: [Int] = []

while let line = readLine() {
    let pieces = line.split(separator: "   ")
    list1.append(Int(pieces[0])!)
    list2.append(Int(pieces[1])!)
}

var counts: [Int: Int] = [:]
for value in list2 {
    counts[value] = (counts[value] ?? 0) + 1
}

var result = 0
for value in list1 {
    result += value * (counts[value] ?? 0)
}

print(result)

File: day01/part2.js
let list1 = [];
let list2 = [];

for await (const line of console) {
    const [v1, v2] = line.split("   ");
    list1.push(Number(v1));
    list2.push(Number(v2));
}

const counts = {};
for (const value of list2) {
    counts[value] = (counts[value] ?? 0) + 1
}

let result = 0;
for (const value of list1) {
    result += value * (counts[value] ?? 0);
}

console.log(result);